controller/controller.go

```diff
@@ -203,18 +199,26 @@ func WebStoreUpdateExtension(w http.ResponseWriter, r *http.Request) {
 
 	w.WriteHeader(http.StatusOK)
 
-	if IsJSONRequest(r.Header.Get("Content-Type")) {
-		w.Header().Set("content-type", "application/json")
-		data, err = json.Marshal(&webStoreResponse)
-	} else {
-		w.Header().Set("content-type", "application/xml")
-		data, err = xml.Marshal(&webStoreResponse)
+	// Always use protocol 3.1 for responses
+	protocolHandler, err := ProtocolFactory.CreateProtocol("3.1")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error creating protocol handler: %v", err), http.StatusInternalServerError)
+		return
 	}
 
+	data, err := protocolHandler.FormatResponse(webStoreResponse, true, contentType)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Error in marshal %v", err), http.StatusInternalServerError)
+		http.Error(w, fmt.Sprintf("Error formatting response: %v", err), http.StatusInternalServerError)
 		return
 	}
+
+	// Set content type
+	if omaha.IsJSONRequest(contentType) {
```

Comment:
Is this needed? or can it simply be:
`w.Header().Set("content-type", contentType)`

Reply:
true - would definitely want to sanitize the value first! we can leave as-is